public/app/plugins/panel/nodeGraph/utils.ts

```diff
@@ -138,12 +143,36 @@ export function processNodes(
     return { nodes: [], edges: [] };
   }
 
+  let hasFixed: boolean | undefined = undefined;
+
   if (nodes) {
     const nodeFields = getNodeFields(nodes);
     if (!nodeFields.id) {
       throw new Error('id field is required for nodes data frame.');
     }
 
+    if (!!nodeFields.fixed_x !== !!nodeFields.fixed_y) {
```

Comment:
Not 100% sure but wouldn't something like this have the same effect:
```
    const hasFixedPositions =
      nodeFields.fixed_x &&
      nodeFields.fixed_x.values.every((v) => Number.isFinite(v)) &&
      nodeFields.fixed_y &&
      nodeFields.fixed_y.values.every((v) => Number.isFinite(v));
    if (!hasFixedPositions) {
      const somePosFilled =
        (nodeFields.fixed_x && nodeFields.fixed_x.values.some((v) => Number.isFinite(v))) ||
        (nodeFields.fixed_y && nodeFields.fixed_y.values.some((v) => Number.isFinite(v)));
      if (somePosFilled) {
        throw new Error('fixed_x and fixed_y values have to all be filled and valid');
      }
    }
```

It skips specific error for field omission but I think we don't need that distinction. Would also be nice to add a test for this.